### DISCUSSION TITLE
Application Autostart 'Trust Desktop Icons' causes post install error if left enabled

### DISCUSSION THREAD

**Discussion by bin-linux** [SpiralLinux/SpiralLinux-project#392] [Q&A]
Spiral XFCE
Rather puzzled finding
`gio: Error setting file metadata: No such file or directory
sha256sum: /home/robin/Desktop/install-debian.desktop: No such file or directory
gio: Error setting file metadata: No such file or directory
chmod: cannot access '/home/robin/Desktop/install-debian.desktop': No such file or directory`
in .xsession-errors so decided to track it down
It is caused by the 'Trust Desktop Icons' entry in Session and Startup>Autostart
This calls /usr/bin/trust-desktop-icons.sh which says:-
`### Make Debian Installer desktop file trusted
sleep 1 ## Stupid hack to avoid occasional race condition
gio set --type=string ~/Desktop/install-debian.desktop metadata::trusted true
gio set --type=string ~/Desktop/install-debian.desktop metadata::xfce-exe-checksum "$(sha256sum ~/Desktop/install-debian.desktop | cut -d ' ' -f1)"
chmod a+x ~/Desktop/install-debian.desktop

### Make Language Support desktop file trusted
gio set --type=string ~/Desktop/language-support.desktop metadata::trusted true
gio set --type=string ~/Desktop/language-support.desktop metadata::xfce-exe-checksum "$(sha256sum ~/Desktop/language-support.desktop | cut -d ' ' -f1)"
chmod a+x ~/Desktop/language-support.desktop`
which is all well and good, but after installation the install-debian.desktop file is no longer there.

Having sorted the languages it therefore makes sense to disable this entry in Autostart.

Ideally a bit of code could be written to deal with this as part of the install - thought I'd just mention it.

**Comment by geckolinux** [comment]
Hi there, you're right that it's not ideal. Unfortunately there's no other way around it with XFCE's annoying concept of "trusting" desktop icons, and there's no way of knowing if/when the user will delete the desktop icon and thus make it no longer necessary for the autostart hack to run.
